google-api-client/src/main/java/com/google/api/client/googleapis/GoogleUtils.java

```diff
@@ -89,20 +89,18 @@ public static synchronized KeyStore getCertificateTrustStore()
   }
 
   private static String getVersion() {
-    String version = GoogleUtils.class.getPackage().getImplementationVersion();
-    // in a non-packaged environment (local), there's no implementation version to read
-    if (version == null) {
-      // fall back to reading from a properties file - note this value is expected to be cached
-      try (InputStream inputStream =
-          GoogleUtils.class.getResourceAsStream("google-api-client.properties")) {
-        if (inputStream != null) {
-          Properties properties = new Properties();
-          properties.load(inputStream);
-          version = properties.getProperty("google-api-client.version");
-        }
-      } catch (IOException e) {
-        // ignore
+    // attempt to read the library's version from a properties file generated during the build
+    // this value should be read and cached for later use
+    String version = "unknown-version";
+    try (InputStream inputStream =
+        GoogleUtils.class.getResourceAsStream("google-api-client.properties")) {
+      if (inputStream != null) {
+        final Properties properties = new Properties();
+        properties.load(inputStream);
+        version = properties.getProperty("google-api-client.version");
```

Comment:
this could be null. You should supply a default here.

Reply:
Done. Moved the default to the return statement while checking for null.

Reply:
Or you could write

```
version = properties.getProperty("google-api-client.version". "unknown-version");
```

up to you.